src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -673,4 +675,21 @@ export class ProgrammingExerciseService {
             .post<ProgrammingExercise>(url, formData, { observe: 'response' })
             .pipe(map((res: EntityResponseType) => this.processProgrammingExerciseEntityResponse(res)));
     }
+
+    getCheckoutDirectoriesForProgrammingLanguage(programmingLanguage: ProgrammingLanguage): Observable<RepositoriesCheckoutDirectoriesDTO> {
+        return this.http
+            .get<RepositoriesCheckoutDirectoriesDTO>(`${this.resourceUrl}/repository-checkout-directories`, {
+                params: {
+                    programmingLanguage,
+                },
+            })
+            .pipe(
+                map((checkoutDirectories: RepositoriesCheckoutDirectoriesDTO) => {
+                    checkoutDirectories.exerciseCheckoutDirectory = addLeadingSlashIfNotPresent(checkoutDirectories.exerciseCheckoutDirectory);
+                    checkoutDirectories.solutionCheckoutDirectory = addLeadingSlashIfNotPresent(checkoutDirectories.solutionCheckoutDirectory);
+                    checkoutDirectories.testCheckoutDirectory = addLeadingSlashIfNotPresent(checkoutDirectories.testCheckoutDirectory);
```

Comment:
Your server code already verifies that there is a slash. Do we need to do this here again?